pkg/antiaffinity/service.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package antiaffinity
+
+import (
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
+	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/services/govmomi/clustermodules"
+)
+
+type service struct{}
+
+func NewService() Service {
+	return service{}
+}
+
+func (s service) Create(ctx *context.AffinityContext, wrapper Wrapper) (string, error) {
```

Comment:
why do you need a new context vs using the `ClusterContext`?

Reply:
Answered this in a comment below, pasting it here again.
> It made sense to have a different context earlier when there was a separate controller. Removing this makes sense now as it is a single controller.